src/main/java/org/spongepowered/api/util/Tristate.java

```diff
@@ -78,6 +95,21 @@ public static Tristate fromBoolean(boolean val) {
         return val ? TRUE : FALSE;
     }
 
+    /**
+     * Returns the appropriate tristate for a given nullable boolean value,
+     * where null equates to {@link Tristate#UNDEFINED}.
+     *
+     * @param val The nullable boolean value
+     * @return The appropriate tristate
+     */
+    public static Tristate fromNullable(@Nullable Boolean val) {
```

Comment:
I feel like this should be named `fromNullableBoolean`.

If we have this, should we also have an `asNullableBoolean` analogue?

Reply:
See, I was thinking about doing `asNullableBoolean`, but what should it return? Returning an actual nullable boolean would make sense, but Sponge prefers Optionals... so maybe OptionalBool? Still then, it feels a little weird.

Reply:
This is one of those rare cases I'd have a nullable return type as a one-to-one mapping of `Boolean`<->`Tristate` - if you can get a `Tristate` from a `Boolean`, it stands to reason that you should be able to go the other way just as simply.

I'll let @Zidane call this one, but I think it should just be a `@Nullable Boolean` return type. There is an extra clue in the name of the method here, if you're not expecting the possibility of it being `null` if you grab the `Boolean` then that's really your lookout.

Reply:
I would prefer, not to have a `asNullableBoolean`, because this class is the replacement for the nullable/unset boolean.

As for the `fromX` methods, I would rather name them both `from` (or `fromBoolean`) with just different parameters. Let the compiler decide which one it should use.

Reply:
> because this class is the replacement for the nullable/unset boolean.

That doesn't change my argument at all. I'm saying the mapping should work both ways in case you then interact with some system that _isn't_ `Tristate` aware - plugins can (and often do) work with third party libraries.

Reply:
Changed `fromNullable` to `fromNullableBoolean`, and added `asNullableBoolean`.

> As for the fromX methods, I would rather name them both from (or fromBoolean) with just different parameters. Let the compiler decide which one it should use.

I'd rather users to explicitly opt-in to using the nullable boolean conversions.